### PR TITLE
Documentation Overhaul

### PR DESCRIPTION
**Pull request by hayesall**
This PR reorganizes the sphinx documentation, abbreviated as follows:

* Home page is leaned down substantially: now it only includes a brief motivation, a gif of the code in action, and links to the bugtracker and issues page. The bottom of the page links to the more in-depth guides.
* Table of contents (on the right of each page), now links to the getting started information and the module-level documentation.
* each module (`corpus`, `parse`, `textprocessing`) has its own page now.
* New "Getting Started" section, including a brief overview of setting up a development environment, installing the software, discussing the relations encoded by the software, and an updated learning task.

Since one of the example files is the U.S. Declaration of Independence, the learning task now reflects the task of determining if a sentence is part of the ["List of Grievances"](https://en.wikibooks.org/wiki/Wikijunior:United_States_Charters_of_Freedom/Declaration_of_Independence#Grievances) against King George III.

---

This PR modifies the `example_files/doi.txt` in order to change some of the end-of-line colon characters to periods. This change was made since the default nltk sentence-tokenizer does not consider colons to be end-of-sentence characters.

### CODE REVIEW

**Comment by hayesall** [comment]
@skinn009 I'd really appreciate a close review on this one, particularly in terms of the written content and whether the tutorial makes sense.

Pending acceptance, squashing these would probably be best.

File: documentation/source/getting_started/03_quickstart.rst [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+  * Existential Semantics:
+
+    * ``sentenceInBlock``: Sentence occurs in a particular block.
+    * ``wordInSentence``: Word occurs in a particular sentence.
+
+  * Low-Level Information about words:
+
+    * ``wordString``: A string representation of a word.
+    * ``partOfSpeechTag``: The word’s part of speech.
```
Comment (skinn009) [inline documentation/source/getting_started/03_quickstart.rst]: How about "The word's part of speech" (POS), determined by the NLTK POS tagger.

File: documentation/source/getting_started/04_learning.rst [hunk 2]
```diff
@@ -0,0 +1,92 @@
+========
+Learning
+========
+
+*This is a brief overview of a learning task. Requirements for a more specific task may vary substantially based on the goals or the data available to you.*
+
+We now have ``bk.txt`` and ``facts.txt`` as a result of the previous step. In order to get show some results, we will construct a toy data set from predicates easily available in the facts file.
+
+The *Delcaration of Independence* contains a set of phrases called the "List of Grievances", where the Founders spell out the 27 violations by King George III.
```
Comment (skinn009) [inline documentation/source/getting_started/04_learning.rst]: Typo Declaration

File: documentation/source/getting_started/03_quickstart.rst (hunk 1 above)
Comment (hayesall) [reply documentation/source/getting_started/03_quickstart.rst]: Good point, let me add that in first thing on Hackathon this Friday.

File: documentation/source/getting_started/04_learning.rst (hunk 2 above)
Comment (hayesall) [reply documentation/source/getting_started/04_learning.rst]: Great catch: "De**lc**aration" =/= "De**cl**aration". Same deal, I'll resolve on Friday.

File: documentation/source/getting_started/03_quickstart.rst (hunk 1 above)
Comment (hayesall) [reply documentation/source/getting_started/03_quickstart.rst]: Resolved in the commit e5a6607e8558dd695f076778231d0ab6b34cfbfe

File: documentation/source/getting_started/04_learning.rst (hunk 2 above)
Comment (hayesall) [reply documentation/source/getting_started/04_learning.rst]: Resolved in e5a6607e8558dd695f076778231d0ab6b34cfbfe

**Pull request merged**